src/main/java/dk/kb/discover/webservice/KBOAuth2Handler.java

```diff
@@ -0,0 +1,498 @@
+        if (tokenParts.length < 2) {
+            log.warn("Received encodedAccessToken string without a dot: '{}'", encodedAccessToken);
+            throw new VerificationException("Unsupported access token (no dot)");
+        }
+
+        JSONObject header = decodeJSONObject(tokenParts[0]);
+        JSONObject payload = decodeJSONObject(tokenParts[1]);
+
+        if (!header.has("kid")) {
+            throw new VerificationException("No key ID (kid) present in access token header");
+        }
+        if (!payload.has("iss")) {
+            throw new VerificationException("No issuer (iss) present in access token payload");
+        }
+        if (mode == MODE.OFFLINE) {
+            log.debug("Authorization mode is " + MODE.OFFLINE + ": Skipping realmURL, publicKey and expiration checks");
+
+            try {
+                JWSInput jws = new JWSInput(encodedAccessToken);
+                return jws.readJsonContent(AccessToken.class);
+            } catch (JWSInputException e) {
+                throw new VerificationException("Failed to parse JWT", e);
+            }
+        }
+
+        String kid = header.get("kid").toString();
+        String realm = getRealm(payload);
+        //  "https://keycloak-keycloak.example.org/auth/realms/brugerbasen"
+        String issuer = baseurl + "/" + realm;
+
+        return TokenVerifier.create(encodedAccessToken, AccessToken.class)
+                // TODO: Figure out why we can't trust iss
+                //.withChecks(new TokenVerifier.RealmUrlCheck(issuer)) // String match only
+                .publicKey(getRealmKey(realm, kid))
+                .verify()
+                .getToken();
+    }
+
+
+    /**
+     * Validate issued date, expiry etc. for the given AccessToken.
+     * @param trusted an AccessToken which has passed the cryptographic validation.
+     * @throws VerificationException if any of the validation steps failed.
+     */
+    private void checkToken(AccessToken trusted) throws VerificationException {
+        final long now = new Date().getTime();
+
+        // Note: Timestamps in token is in seconds since Epoch. Date().getTime is milliseconds
+
+        if (trusted.isExpired()) {
+            long overtime = now/1000 - trusted.getExpiration();
+            throw new VerificationException("AccessToken expired (" + overtime + " seconds too old)");
+        }
+
+        if (!trusted.isNotBefore(0)) {
+            long missing = trusted.getNotBefore() - now/1000;
+            throw new VerificationException("AccessToken not valid before " + missing + " seconds has passed");
+        }
+
+        if (trusted.getIssuedAt() > now) {
+            long missing = trusted.getIssuedAt()/1000 - now;
+            log.warn("Received trusted AccessToken issued {}} seconds in the future (epoch seconds = {})",
+                     missing, trusted.getIssuedAt());
+            throw new VerificationException("AccessToken issued " + missing + " seconds in the future");
+        }
```

Comment:
trusted.get methods here are all deprecated. IntelliJ says that they can be swapped 1-to-1 with .getExp(), .getNbf() and .getIat could we test these non-deprecated methods and update them now and add a single comment on top of each of them as the new methods are abbreviated. Comments like // Check Expiration // Check Not before and // Check Issues at

Reply:
Fixed. Also tested token still expire correct